lib/new_api_prototype.dart

```diff
@@ -0,0 +1,895 @@
+    'Exactly one of the inputs must be null',
+  );
+  return (value) {
+    final T actualN;
+    if (n != null) {
+      actualN = n;
+    } else if (dynN != null) {
+      actualN = dynN();
+    } else {
+      throw TypeError();
+    }
+    return value >= actualN
+        ? null
+        : greaterThanOrEqualMessage?.call(actualN) ??
+            'Value must be greater than or equal to $actualN';
+  };
+}
+
+Validator<T> lessThan<T extends num>(T? n,
+    [T Function()? dynN, String Function(num)? lessThanMessage]) {
+  assert(
+    n != null && dynN == null || n == null && dynN != null,
+    'Exactly one of the inputs must be null',
+  );
+  return (value) {
+    final T actualN;
+    if (n != null) {
+      actualN = n;
+    } else if (dynN != null) {
+      actualN = dynN();
+    } else {
+      throw TypeError();
+    }
+    return value < actualN
+        ? null
+        : lessThanMessage?.call(actualN) ?? 'Value must be less than $actualN';
+  };
+}
+
+Validator<T> lessThanOrEqual<T extends num>(T? n,
+    [T Function()? dynN, String Function(num)? lessThanOrEqualMessage]) {
+  assert(
+    n != null && dynN == null || n == null && dynN != null,
+    'Exactly one of the inputs must be null',
+  );
+  return (value) {
+    final T actualN;
+    if (n != null) {
+      actualN = n;
+    } else if (dynN != null) {
+      actualN = dynN();
+    } else {
+      throw TypeError();
+    }
+    return value <= actualN
+        ? null
+        : lessThanOrEqualMessage?.call(actualN) ??
+            'Value must be less than or equal to $actualN';
+  };
+}
+
+Validator<T> between<T extends num>(T? min, T? max,
+    {T Function()? dynMin,
+    T Function()? dynMax,
+    bool leftInclusive = true,
+    bool rightInclusive = true,
+    String Function(num min, num max)? betweenMessage}) {
+  assert(
+    min != null && dynMin == null || min == null && dynMin != null,
+    'Exactly one of the min inputs must be null',
+  );
+  assert(
+    max != null && dynMax == null || max == null && dynMax != null,
+    'Exactly one of the max inputs must be null',
+  );
+  return (value) {
+    final T actualMin;
+    final T actualMax;
+    if (min != null) {
+      actualMin = min;
+    } else if (dynMin != null) {
+      actualMin = dynMin();
+    } else {
+      throw TypeError();
+    }
+    if (max != null) {
+      actualMax = max;
+    } else if (dynMax != null) {
+      actualMax = dynMax();
+    } else {
+      throw TypeError();
+    }
+    return (leftInclusive ? value >= actualMin : value > actualMin) &&
+            (rightInclusive ? value <= actualMax : value < actualMax)
+        ? null
+        : betweenMessage?.call(actualMin, actualMax) ??
+            'Value must be greater than ${leftInclusive ? 'or equal to ' : ''}$actualMin and less than ${rightInclusive ? 'or equal to ' : ''}$actualMax';
+  };
+}
+
+const gt = greaterThan;
+const gtE = greaterThanOrEqual;
+const lt = lessThan;
+const ltE = lessThanOrEqual;
+const bw = between;
```

Comment:
Also about this shorcuts and validators name in general, several cases of conflicts may arise when using it in real applications because the validator name is the name of local variables or other packages resources

Reply:
I changed them. Please, check if it is ok.